File: apps/frontend/src/types/impact-analysis.ts
export interface ImpactAnalysis {
  id: string;
  workflow_id: string;
  business_impact: number;
  technical_complexity: number;
  urgency: number;
  confidence: number;
  risk_score: number;
  opportunity_score: number;
  final_priority: string;
  impact_breakdown: Record<string, number>;
  recommendation?: string | null;
  created_at: string;
}

File: apps/backend/app/schemas/impact_analysis.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ImpactAnalysisRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    workflow_id: UUID
    business_impact: float
    technical_complexity: float
    urgency: float
    confidence: float
    risk_score: float
    opportunity_score: float
    final_priority: str
    impact_breakdown: dict[str, Any] = Field(default_factory=dict)
    recommendation: str | None = None
    created_at: datetime
